src/compiler/checker.ts

```diff
@@ -13678,8 +13693,6 @@ namespace ts {
                     spread = getSpreadType(spread, createObjectLiteralType());
                 }
                 if (spread.flags & TypeFlags.Object) {
-                    // only set the symbol and flags if this is a (fresh) object type

```

Comment:
this comment should stay, although maybe it should explain that sometimes getSpreadType returns anyType (for example).